exercises/practice/run-length-encoding/RunLengthEncodingTest.php

```diff
@@ -31,59 +9,154 @@ public static function setUpBeforeClass(): void
         require_once 'RunLengthEncoding.php';
     }
 
+    /**
+     * uuid: ad53b61b-6ffc-422f-81a6-61f7df92a231
+     * @testdox empty string
+     */
     public function testEncodeEmptyString(): void
     {
-        $this->assertEquals('', encode(''));
+        $this->assertEquals(
+            '',
+            encode('')
+        );
     }
 
-    public function testEncodeSingleCharactersOnly(): void
+    /**
+     * uuid: 52012823-b7e6-4277-893c-5b96d42f82de
+     * @testdox single characters only are encoded without count
+     */
+    public function testEncodeSingleCharactersOnlyAreEncodedWithoutCount(): void
     {
-        $this->assertEquals('XYZ', encode('XYZ'));
+        $this->assertEquals(
+            'XYZ',
+            encode('XYZ')
+        );
     }
 
-    public function testDecodeEmptyString(): void
+    /**
+     * uuid: b7868492-7e3a-415f-8da3-d88f51f80409
+     * @testdox string with no single characters
+     */
+    public function testEncodeStringWithNoSingleCharacters(): void
     {
-        $this->assertEquals('', decode(''));
+        $this->assertEquals(
+            '2A3B4C',
+            encode('AABBBCCCC')
+        );
     }
 
-    public function testDecodeSingleCharactersOnly(): void
+    /**
+     * uuid: 859b822b-6e9f-44d6-9c46-6091ee6ae358
+     * @testdox single characters mixed with repeated characters
+     */
+    public function testEncodeSingleCharactersMixedWithRepeatedCharacters(): void
     {
-        $this->assertEquals('XYZ', decode('XYZ'));
+        $this->assertEquals(
+            '12WB12W3B24WB',
+            encode('WWWWWWWWWWWWBWWWWWWWWWWWWBBBWWWWWWWWWWWWWWWWWWWWWWWWB')
+        );
     }
 
-    public function testEncodeSimple(): void
+    /**
+     * uuid: 1b34de62-e152-47be-bc88-469746df63b3
+     * @testdox multiple whitespace mixed in string
+     */
+    public function testEncodeMultipleWhitespaceMixedInString(): void
     {
-        $this->assertEquals('2A3B4C', encode('AABBBCCCC'));
+        $this->assertEquals(
+            '2 hs2q q2w2 ',
+            encode('  hsqq qww  ')
+        );
     }
 
-    public function testDecodeSimple(): void
+    /**
+     * uuid: abf176e2-3fbd-40ad-bb2f-2dd6d4df721a
+     * @testdox lowercase characters
+     */
+    public function testEncodeLowercaseCharacters(): void
     {
-        $this->assertEquals('AABBBCCCC', decode('2A3B4C'));
+        $this->assertEquals(
+            '2a3b4c',
+            encode('aabbbcccc')
+        );
     }
 
-    public function testEncodeWithSingleValues(): void
+    /**
+     * uuid: 7ec5c390-f03c-4acf-ac29-5f65861cdeb5
+     * @testdox empty string
```

Comment:
Here the prefix changes:

```suggestion
     * @testdox Run-length decode a string - empty string
```